src/Core/Streaming/Watermark.h

```diff
@@ -9,5 +9,48 @@ namespace Streaming
 /// The 1970-01-01 00:00:00 is Int64 value 0, earlier times are negative, so we need to use signed Int64 as watermark
 constexpr Int64 INVALID_WATERMARK = std::numeric_limits<Int64>::min();
 constexpr Int64 TIMEOUT_WATERMARK = std::numeric_limits<Int64>::max();
+
+
+enum class WatermarkStrategy
+{
+    Unknown,
+
+    /// No watermark
+    None,
+
+    /// Watermark is generated based on each event, this is used for `EMIT AFTER WATERMARK WITHOUT DELAY`
```

Comment:
For every event ? What's the use case ?

Reply:
For session window, we need events timestamp is in ascending, which is required by session start/end